### PR TITLE
[Feature] : Horizontal bar visualization implementation & 2way sync

### PR DESCRIPTION
**Pull request by DipraAich**
### Description
Merging horizontal and vertical bar chart visualizations
Implementing 2way sync in horizontal bar chart

### Issues Resolved
[#885](https://github.com/opensearch-project/observability/issues/885)

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/utils/utils.tsx [hunk 1]
```diff
@@ -363,8 +363,7 @@ export const fetchConfigObject = (editor: string, propsOptions: any) => {
 export const getTooltipHoverInfo = ({ tooltipMode, tooltipText }: GetTooltipHoverInfoType) => {
   if (tooltipMode === 'hidden') {
     return 'none';
-  }
-  if (tooltipText === undefined) {
+  } else if (tooltipText === undefined) {
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/utils/utils.tsx]: You may not require else here, as the if condition above would anyway return a value.

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx [hunk 2]
```diff
@@ -62,17 +67,19 @@ export const Bar = ({ visualizations, layout, config }: any) => {
     dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== visMetaData.showlegend
   );
   const legendPosition = dataConfig?.legend?.position || visMetaData.legendposition;
-  const labelSize = dataConfig?.chartStyles?.labelSize || DEFAULT_LABEL_SIZE;
 
+  visualizations.data?.rawVizData?.dataConfig?.metrics
+    ? visualizations.data?.rawVizData?.dataConfig?.metrics
```
Comment (spattnaik) [inline dashboards-observability/public/components/visualizations/charts/bar/bar.tsx]: You may not need any question marks on line 72. As the if condition is satisfied, that means metrics do exist.

**Review by spattnaik — APPROVED** [review]
Please take care of the minor comments I have given

File: dashboards-observability/public/components/event_analytics/utils/utils.tsx (hunk 1 above)
Comment (DipraAich) [reply dashboards-observability/public/components/event_analytics/utils/utils.tsx]: done

File: dashboards-observability/public/components/visualizations/charts/bar/bar.tsx (hunk 2 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/bar/bar.tsx]: done

**Pull request closed without merging**